frontend/src/lib/dataFrameProto.ts

```diff
@@ -237,11 +259,20 @@ export function dataFrameGet(df: any, col: any, row: any): DataFrameCell {
       ? customDisplayValue
       : tableGet(df.get("data"), col - headerCols, row - headerRows)
 
+  let styles =
+    tableStyleGetCSS(df.get("style"), col - headerCols, row - headerRows) || {}
+
+  // align data columns to left when the content is not a number
+  if (typeof contents !== "number") {
```

Comment:
I'm not entirely sure if this will fix it, but I think we'll likely need to look at the raw data as opposed to `customDisplayValue` so that alignments are correct in all cases.

Here's a snapshot diff that cypress flagged that left-aligns numbers that should be right-aligned. The old snapshot is shown on the left and new one on the right.
![table-formatted-cells-dark diff](https://user-images.githubusercontent.com/3144420/115478134-5121d600-a1fa-11eb-9325-2d8f5203d2c9.png)

There are similar snapshot failures from that same e2e test (`e2e/scripts/st_table_styling.py`) that are similarly misaligned, and they all happen to be styled dataframes.

Reply:
Done